### ISSUE TITLE
Derivation of Flow matching loss used in paper

### ISSUE THREAD

**Issue by DDDaxing** [FoundationVision/FlashVideo#16]
Hi authors, in your paper, the loss target and noise_latent (model input) came to be 
`target = z_HR - z_LR`， and the `noise_latent = (1-t)z_LR + t * z_HR`

While in the original flow matching paper (or normally in flux related model), the loss and and noise_latent is:
`target = noise - z_HR`， and the `noise_latent = (1-t)z_HR + t * noise`

Although the `z_LR` can be viewed as a HR image with some noise, there is still gap between `z_LR` and random noise. Will the flow matching loss still hold that way and follow the linearity of the ReFlow trajectory when the starting point is LR image rather than noise? 

Can you please provide the derivation for the loss? Thank you.

**Comment by jshilong** [comment]
Thank you for your interest in our work. We would like to clarify a few points, as there are some aspects of your questions that seem to reflect certain misunderstandings:

1. **Linearity of the ReFlow trajectory**: First, it's important to note that "ReFlow" refers to a distinct concept that is unrelated to our study. Our study does not discuss or utilize any concepts associated with ReFlow.

2. **Flow matching target make sure the trajectory linearity**: The idea that flow matching guarantees a straight trajectory is a common misconception. Flow matching only ensures straightness between each sampled pair in the training. Ultimately, when trained across the entire dataset, the resulting trajectory is the average of all sampled pairs. Therefore, the overall trajectory is typically not linear. But it is indeed straighter than the previously commonly used DDPM.

3. Initial distribution in flow matching/rectified flow: The original flow matching and rectified flow formulations do not assume that the initial distribution must be Gaussian. This can be verified by reviewing the original papers. 


Feel free to reach out if you have further questions or would like to discuss this in more detail：）

**Comment by DDDaxing** [comment]
> Thank you for your interest in our work. We would like to clarify a few points, as there are some aspects of your questions that seem to reflect certain misunderstandings:
> 
> 1. **Linearity of the ReFlow trajectory**: First, it's important to note that "ReFlow" refers to a distinct concept that is unrelated to our study. Our study does not discuss or utilize any concepts associated with ReFlow.
> 2. **Flow matching target make sure the trajectory linearity**: The idea that flow matching guarantees a straight trajectory is a common misconception. Flow matching only ensures straightness between each sampled pair in the training. Ultimately, when trained across the entire dataset, the resulting trajectory is the average of all sampled pairs. Therefore, the overall trajectory is typically not linear. But it is indeed straighter than the previously commonly used DDPM.
> 3. **Initial distribution in flow matching/rectified flow**: The original flow matching and rectified flow formulations do not assume that the initial distribution must be Gaussian. This can be verified by reviewing the original papers.
> 
> Feel free to reach out if you have further questions or would like to discuss this in more detail：）

Thank you for your quick reply! Got your ideas. Right, the trajectory will not be straight in the training. Seems you directly use the "linear interpolation" idea to implement in the training. In this recent paper (https://arxiv.org/pdf/2502.01993), the authors show that distribution shift exists between the LR image distribution and the real data distribution, therefore they use two trajectories in the triangle to solve the shift manner. I'm wondering if your results echo their findings?

BTW, I found that you also augment the LR in latent by using a square sum with Gaussian noise. It looks very similar to the noise adding format as that in DDPM. Have you tested that this step is essential? And maybe using the sum format(which is used in Flux) also works too?

![Image](https://github.com/user-attachments/assets/aaadfc3a-b34c-4a5d-87d7-85d214618dc2)

**Comment by jshilong** [comment]
1. I have not read this paper, but it seems they aim to address the mismatch between pretraining and the current task. However, our approach trains from scratch, and we have not observed this issue in our experiments.

2. Adding latent noise is indeed necessary. We have emphasized this point in our method as well as in the ablation studies. Direct linear fuse the noise and latent should theoretically work as well; the key factor is ensuring that the intensity is appropriately set.
